### PR TITLE
New article regarding disabling weak cipher suites and enabling HSTS in OpenShift clusters

### PR DESCRIPTION
**Pull request by benbacon**

### CODE REVIEW

File: articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+There are no weak ciphers supported by the OpenShift control plane (API) since by default it supports TLS 1.2 and 1.3 cipher suites with forward secrecy.
+
+## Enable HSTS
+
+HSTS (HTTP Strict Transport Security) prevents MITM attacks such as protocol downgrading by setting a header within HTTPS responses. This header (`Strict-Transport-Security`) is cached by a browser and ensures that future requests to the same host always use HTTPS as opposed to HTTP.
+
+> [!NOTE]
+> It is currently not possible to enable HSTS for the OpenShift v3.x web console or the OpenShift v4.x API server.
+
+HSTS can be enabled on secure routes by using the following command:
```
Comment (gellner) [inline articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md:73]: We should include a warning that says to only do this on secure routes that have valid certificates. I guess all prod clusters will produce domain names which are valid with the default domains for routes, but we'd need to explain that secure custom-domain-name routes need a valid cert against the route?

Comment (benbacon) [reply articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md:76]: @gellner just pushed an update containing a warning

File: articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Enable HSTS
+
+HSTS (HTTP Strict Transport Security) prevents MITM attacks such as protocol downgrading by setting a header within HTTPS responses. This header (`Strict-Transport-Security`) is cached by a browser and ensures that future requests to the same host always use HTTPS as opposed to HTTP.
+
+> [!NOTE]
+> It is currently not possible to enable HSTS for the OpenShift v3.x web console or the OpenShift v4.x API server.
+
+> [!WARNING]
+> HSTS should only be enabled on routes which currently have a valid certificate; should the certificate become invalid the route will be inaccessible, due to how browsers' HSTS mechanism works. Particular caution should be exercised if you are using a non-standard hostname (i.e. not ending in `<your-cluster-hostname>.ukcloud.com`) for a route - in this case you should ensure a valid certificate is contained within the route spec.
```
Comment (koneill94) [inline articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md:74]:
```suggestion
> HSTS should only be enabled on routes which currently have a valid certificate; should the certificate become invalid the route will be inaccessible, due to how browsers' HSTS mechanism works. Particular caution should be exercised if you are using a non-standard hostname (i.e. not ending in `<your-cluster-hostname>.ukcloud.com`) for a route - in this case you should ensure a valid certificate is contained within the route spec for edge and re-encrypt routes or within the container if using the pass-through TLS encryption method.
```

LGTM minor suggested change about where certs are contained

File: articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+## Enable HSTS
+
+HSTS (HTTP Strict Transport Security) prevents MITM attacks such as protocol downgrading by setting a header within HTTPS responses. This header (`Strict-Transport-Security`) is cached by a browser and ensures that future requests to the same host always use HTTPS as opposed to HTTP.
+
+> [!NOTE]
+> It is currently not possible to enable HSTS for the OpenShift v3.x web console or the OpenShift v4.x API server.
+
+> [!WARNING]
+> HSTS should only be enabled on routes which currently have a valid certificate; should the certificate become invalid the route will be inaccessible, due to how browsers' HSTS mechanism works. Particular caution should be exercised if you are using a non-standard hostname (i.e. not ending in `<your-cluster-hostname>.ukcloud.com`) for a route - in this case you should ensure a valid certificate is contained within the route spec for edge and re-encrypt routes or within the container if using the pass-through TLS encryption method.
```
Comment (suehighmoor) [inline articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md:74]: The last bit of this (starting  ...route spec for edge...) didn't really make sense to me (as much of any of this stuff actually makes sense to me). Can you maybe reword to make it clearer? I don't understand it enough to make a suggestion :(

File: articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+toc_title: Disable weak cipher suites and enable HSTS
+toc_fullpath: How To/oshift-how-disable-weak-cipher-suites-enable-hsts.md
+toc_mdlink: oshift-how-disable-weak-cipher-suites-enable-hsts.md
+---
+
+# How to disable weak cipher suites and enable HSTS
+
+## Overview
+
+This article outlines steps to increase the security of externally exposed routes from your OpenShift cluster, through the use of more secure cipher suites and HSTS.
```
Comment (andrew-garner) [inline articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md:22]: Could use "HTTP Strict Transport Security (HSTS)" in the overview.

File: articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+By default, the OpenShift control plane (web console) only supports TLS 1.2 ciphers, although these do not provide forward secrecy.
+
+We can restrict the ciphers to only those that provide forward secrecy. To do this, raise a Service Request with the UKCloud OpenShift Support team via the [My Calls](https://portal.skyscapecloud.com/support/ivanti) section of the UKCloud Portal.
+
+### UKCloud for OpenShift v4.x
+
+#### Data plane
+
+There are no weak ciphers supported by the OpenShift `IngressController` since by default it supports TLS 1.2 cipher suites with forward secrecy. It is possible to further restrict the ciphers by modifying the default `IngressController` object, specifically the value of `IngressController.spec.tlsSecurityProfile`:
```
Comment (andrew-garner) [inline articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md:56]: Could we say what the value should be? The command below will just allow the customer to edit the config, I'm not clear what the value should be for the IngressController.spec.tlsSecurityProfile

**Review by andrew-garner — CHANGES_REQUESTED** [review]
Minor comments only.

**Comment by suehighmoor** [comment]
@benbacon where are we at with this?

File: articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md (hunk 5 above)
Comment (benbacon) [reply articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md:56]: I've added an example is this clearer?

**Review by gellner — APPROVED** [review]
LGTM - probably time to get this published

**Review by suehighmoor — APPROVED** [review]
LGTM

**Comment by suehighmoor** [comment]
@andrew-garner - as you'd commented on this, please could you just confirm that you're happy that your comments have been addressed and approve?

**Review by andrew-garner — APPROVED** [review]
Yes, thanks. Approved.
